jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/io/AbstractWebSocketConnection.java

```diff
@@ -631,15 +632,16 @@ public SuspendToken suspend()
     }
 
     @Override
-    public String dump()
-    {
-        return Dumpable.dump(this);
+    public String dumpSelf() {
+        return String.format("%s@%x", this.getClass().getSimpleName(), hashCode());
     }
 
-    @Override
-    public void dump(Appendable out, String indent) throws IOException
-    {
-        out.append(toString()).append(System.lineSeparator());
+    public void dump(Appendable out, String indent) throws IOException {
+        Object endpRef = toConnectionString();
+        EndPoint endp = getEndPoint();
+        if(endp instanceof ChannelEndPoint)
+            endpRef = ((ChannelEndPoint) endp).toEndPointString();
+        Dumpable.dumpObjects(out, indent, this, endpRef, ioState, flusher, generator, parser);
```

Comment:
Why not just passing the `endp` to `dumpObjects`?
The `instanceof` will fail for proxied `EndPoint` so seems a special case where it should not be needed (just pass the `EndPoint` and it will dump itself).